lib/Resolver/ResolverInterface.php

```diff
@@ -15,18 +14,12 @@
  *  @license    http://www.pimcore.org/license     GPLv3 and PCL
  */
 
-namespace Pimcore\Model\Document\TypeDefinition\Loader;
-
-use Pimcore\Loader\ImplementationLoader\LoaderInterface;
-use Pimcore\Model\Document;
+namespace Pimcore\Resolver;
 
-interface TypeLoaderInterface extends LoaderInterface
+interface ResolverInterface
 {
     /**
-     * @param string $name
-     * @param array $params
-     *
-     * @return Document
+     * Returns FQCN of class
      */
-    public function build(string $name, array $params = []): Document;
+    public function resolve(string $class, string $type): ?string;
```

Comment:
```suggestion
    public function resolve(string $name, string $type): ?string;
```
Maybe this should stay `$name` because actually it's just an id-string, not a class, right?

Reply:
This is true, it could be a class, but in general, it is better to call it name.